### PR TITLE
[Merged by Bors] - Feature/rework storage (replace local- with HDFS-storage for deep storage)

### LINKED ISSUE #124: Investigate Necessity of the Local Deep Storage Label
is this label really necessary?

### LINKED ISSUE #187: Enable HDFS for deep storage
As a user of druid services I want to be able to use HDFS for deep storage. 

Using local storage for deep storage only works if the storage location is reachable by all relevant druid services (middle-manager, historical server).

- This would be possible with e.g. Persistent Volume and Persistent Volume Claims, but would only realistically be used for prototype/demo cases
- a better solution is to use HDFS and deactivate local-deep-storage as we should offer HDFS **_anyway_**, and it lends itself more easily to integration testing (using our own operator).

### PR DESCRIPTION
**Pull request by adwk67**
## Description

- local storage has been replaced with HDFS storage for Druid deep-storage
  - this adds a storage feature and also makes data ingestion easier to integration test
  - this removes the local deep storage label as it is no longer needed

Closes #124.
Closes #187.

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,14 +9,18 @@ All notable changes to this project will be documented in this file.
 - Reconciliation errors are now reported as Kubernetes events ([#178]).
 - Use cli argument `watch-namespace` / env var `WATCH_NAMESPACE` to specify
   a single namespace to watch ([#183]).
+- Local backend storage (deep-storage) has been replaced with HDFS-storage ([#187]).
```
Comment (maltesander) [inline CHANGELOG.md:12]: This is breaking (for the CRD at least)?

File: rust/operator-binary/src/druid_controller.rs [hunk 2]
```diff
@@ -545,7 +495,7 @@ fn build_rolegroup_statefulset(
             },
             service_name: rolegroup_ref.object_name(),
             template: pb.build_template(),
-            volume_claim_templates: Some(pvcs),
+            volume_claim_templates: Some(vec![]),
```
Comment (maltesander) [inline rust/operator-binary/src/druid_controller.rs:498]: Can be removed completly?

**Review by maltesander — COMMENTED** [review]
LGTM. 2 minor comments. The questions is about https://github.com/stackabletech/druid-operator/issues/188 if we want to include it now or later. @lfrancke @soenkeliebau what do you think?

File: rust/operator-binary/src/druid_controller.rs (hunk 2 above)
Comment (adwk67) [reply rust/operator-binary/src/druid_controller.rs]: done

File: CHANGELOG.md (hunk 1 above)
Comment (adwk67) [reply CHANGELOG.md]: marked as such in changelog - done

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by adwk67** [comment]
bors merge

**Comment by soenkeliebau** [comment]
Bit late to the party here, sorry for opening up the discission again.

Wouldn't  the label become not needed as well, if we put the local storage directory on a mounted persistant volume? And as long as we disallow having more than one node with the role that uses the persistant storage (not sure which one that is, sorry) if local storage is configured then that should not create issues?

I am asking because "local" storage might be useful for "minimal" deployments that are just used for testing purposes..

**Comment by adwk67** [comment]
Yes, that's certainly true (that local storage with a mounted volume would make sense for test installations) but how would we police that? The original issue was that using local storage caused implicit/almost-silent failures (in the case of ingestion), and a clean implementation - with PersistentVolumeClaims and what-not - was going to add a lot of complexity. I agree that spinning up the HDFS stuff for demos takes more time than one would want.

**Pull request closed without merging**